service/src/main/java/uk/nhs/adaptors/gp2gp/gpc/GpcPatientDataHandler.java

```diff
@@ -0,0 +1,42 @@
+package uk.nhs.adaptors.gp2gp.gpc;
+
+import static java.time.Instant.now;
+
+import static uk.nhs.adaptors.gp2gp.gpc.GpcFileNameConstants.GPC_STRUCTURED_FILE_EXTENSION;
+
+import java.util.Optional;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import uk.nhs.adaptors.gp2gp.ehr.EhrExtractStatus;
+import uk.nhs.adaptors.gp2gp.ehr.EhrExtractStatusRepository;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class GpcPatientDataHandler {
```

Comment:
I think this class name is misleading. It is not actually handling patient data all!

This is our service / repository class for making changes to our ehrExtractStatus database collection.

Maybe we need a top-level package named 'status'? EhrExtractStatus, its repostiory can live there. Rename this `EhrExtractStatusService` which will encapsulate all repository / mongotemplate operation?